File: ui/src/types/ai.ts
export interface AIProviderProfile {
  id: number
  name: string
  provider: 'gemini' | 'openai' | 'azure' | 'custom'
  baseUrl: string
  defaultModel: string
  apiKey?: string
  isSystem: boolean
  isEnabled: boolean
  allowUserOverride: boolean
  allowedModels: string[]
  createdAt: string
  updatedAt: string
}

export interface AIGovernanceSettings {
  allow_user_keys: string
  force_user_keys: string
  allow_user_override: string
}

export interface AISettings {
  id?: number
  userID: number
  profileID: number
  apiKey: string
  modelOverride: string
  isActive: boolean
  isDefault?: boolean
  createdAt?: string
  updatedAt?: string
}

export interface AIChatSession {
  id: string
  userID: number
  title: string
  createdAt: string
  updatedAt: string
  messages?: AIChatMessage[]
}

export interface AIChatMessage {
  id?: number
  sessionID?: string
  role: 'system' | 'user' | 'assistant' | 'tool'
  content: string
  toolCalls?: string // JSON string
  toolID?: string
  createdAt: string
}

export interface ChatRequest {
  sessionID: string
  message: string
  model?: string
}

export interface ChatResponse {
  sessionID: string
  message: string
}
export interface AIModelsResponse {
  models: string[]
  default: string
  provider: string
  message?: string
}

File: pkg/model/ai.go
package model

import (
	"time"

	"github.com/pixelvide/kube-sentinel/pkg/common"
	"gorm.io/gorm"
)

type AIProviderProfile struct {
	Model
	Name              string      `json:"name"`
	Provider          string      `json:"provider"` // "gemini", "openai", "azure", "custom"
	BaseURL           string      `json:"baseUrl"`
	DefaultModel      string      `json:"defaultModel"`
	APIKey            string      `json:"apiKey" gorm:"type:text"` // Global key for this profile
	IsSystem          bool        `json:"isSystem"`
	IsEnabled         bool        `json:"isEnabled" gorm:"default:true"` // If false, profile is hidden from users
	AllowUserOverride bool        `json:"allowUserOverride"`             // If true, users can provide their own key
	AllowedModels     SliceString `json:"allowedModels"`                 // Comma-separated in DB, array in JSON
}

func (AIProviderProfile) TableName() string {
	return common.GetAppTableName("ai_provider_profiles")
}

type AISettings struct {
	Model
	UserID        uint   `json:"userID" gorm:"uniqueIndex:idx_user_profile"`
	ProfileID     uint   `json:"profileID" gorm:"uniqueIndex:idx_user_profile"`
	APIKey        string `json:"apiKey"`
	ModelOverride string `json:"modelOverride"`
	IsActive      bool   `json:"isActive"`
	IsDefault     bool   `json:"isDefault"` // If true, this is the user's default AI profile
}

func (AISettings) TableName() string {
	return common.GetAppTableName("k8s_ai_settings")
}

type AIChatSession struct {
	ID        string          `json:"id" gorm:"primaryKey"` // UUID
	UserID    uint            `json:"userID" gorm:"index"`
	Title     string          `json:"title"`
	CreatedAt time.Time       `json:"createdAt"`
	UpdatedAt time.Time       `json:"updatedAt"`
	DeletedAt gorm.DeletedAt  `json:"deletedAt" gorm:"index"`
	Messages  []AIChatMessage `json:"messages" gorm:"foreignKey:SessionID;constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
}

func (AIChatSession) TableName() string {
	return common.GetAppTableName("k8s_ai_chat_sessions")
}

type AIChatMessage struct {
	ID        uint           `json:"id" gorm:"primaryKey"`
	SessionID string         `json:"sessionID" gorm:"index"`
	Role      string         `json:"role"`                                 // "system", "user", "assistant", "tool"
	Content   string         `json:"content"`                              // Text content
	ToolCalls string         `json:"toolCalls,omitempty" gorm:"type:text"` // JSON encoded tool calls
	ToolID    string         `json:"toolID,omitempty"`                     // For tool messages
	CreatedAt time.Time      `json:"createdAt"`
	DeletedAt gorm.DeletedAt `json:"deletedAt" gorm:"index"`
}

func (AIChatMessage) TableName() string {
	return common.GetAppTableName("k8s_ai_chat_messages")
}
